test/mjsunit/wasm/asm-wasm.js

```diff
@@ -2,19 +2,44 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-function foo() {
+function IntTest() {
   "use asm";
   function sum(a, b) {
     a = a|0;
-    c = b|0 + 1;
-    return a + c + 1;
+    b = b|0;
+    var c = (b + 1)|0
+    return (a + c + 1)|0;
   }
 
   function caller() {
-    return sum(77, 22);
+    return sum(77,22) | 0;
   }
 
   return {caller: caller};
 }
 
-assertEquals(101, WASM.asmCompileRun(foo.toString()));
+assertEquals(101, WASM.asmCompileRun(IntTest.toString()));
+
+function Float64Test() {
+  "use asm";
+  function sum(a, b) {
+    a = +a;
+    b = +b;
+    return +(a + b);
+  }
+
+  function caller() {
+    var a = +sum(70.0,10.0);
+    var ret = 0|0;
+    if (a == 80.0) {
+      ret = 1|0;
+    } else {
+      ret = 0|0;
+    }
+    return ret|0;
+  }
+
+  return {caller: caller};
+}
+
+assertEquals(1, WASM.asmCompileRun(Float64Test.toString()));
```

Comment:
The function asmCompileRun can only return 1 at the moment. Will add new function in next commit.